### ISSUE TITLE
Error when talking to some profession trainers

### ISSUE THREAD

**Issue by fusionpit** [fusionpit/ClassTrainerPlus#7]
Via [ziogio70 on curseforge](https://www.curseforge.com/wow/addons/class-trainer-plus?comment=3)

```
1x ClassTrainerPlus\ClassTrainerPlus-0.5-beta.lua:434: attempt to index local 'service' (a nil value)
ClassTrainerPlus\ClassTrainerPlus-0.5-beta.lua:434: in function `ClassTrainerPlus_SetSelection'
ClassTrainerPlus\ClassTrainerPlus-0.5-beta.lua:235: in function <ClassTrainerPlus\ClassTrainerPlus.lua:221>

Locals:
id = 2
showIgnored = true
serviceName = nil
serviceSubText = nil
serviceType = nil
isExpanded = nil
service = nil
(*temporary) = <table> {
visibleServices = 0
VisibleAvailableServiceIds = <function> defined @ClassTrainerPlus\TrainerServices.lua:73
_byServiceId = <table> {
}
IsSelected = <function> defined @ClassTrainerPlus\TrainerServices.lua:138
Update = <function> defined @ClassTrainerPlus\TrainerServices.lua:131
totalServices = 0
showIgnored = true
_updateCandidates = <function> defined @ClassTrainerPlus\TrainerServices.lua:14
availableCost = 0
GetFirstVisibleNonHeaderService = <function> defined @ClassTrainerPlus\TrainerServices.lua:145
_byPosition = <table> {
}
GetNextAvailableServiceId = <function> defined @ClassTrainerPlus\TrainerServices.lua:152
_filter = ""
GetService = <function> defined @ClassTrainerPlus\TrainerServices.lua:167
GetServiceAtPosition = <function> defined @ClassTrainerPlus\TrainerServices.lua:164
SetFilter = <function> defined @ClassTrainerPlus\TrainerServices.lua:83
_candidates = <table> {
}
allHeadersCollapsed = true
ApplyFilter = <function> defined @ClassTrainerPlus\TrainerServices.lua:88
}
(*temporary) = 2
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'service' (a nil value)"
ctp = <table> {
TrainerServices = <table> {
}
L = <table> {
}
Abilities = <table> {
}
RealSpellNameMap = <table> {
}
}
CollapseTrainerSkillLine = <function> defined =[C]:-1
ExpandTrainerSkillLine = <function> defined =[C]:-1
SelectTrainerService = <function> defined =[C]:-1
GetTrainerServiceIcon = <function> defined =[C]:-1
GetTrainerServiceLevelReq = <function> defined =[C]:-1
IsTrainerServiceLearnSpell = <function> defined =[C]:-1
UnitLevel = <function> defined =[C]:-1
format = <function> defined =[C]:-1
GetTrainerServiceSkillReq = <function> defined =[C]:-1
GetTrainerServiceNumAbilityReq = <function> defined =[C]:-1
GetTrainerServiceAbilityReq = <function> defined =[C]:-1
_ = nil
GetTrainerServiceInfo = <function> defined =[C]:-1
GetTrainerServiceCost = <function> defined =[C]:-1
GetMoney = <function> defined =[C]:-1
SetMoneyFrameColor = <function> defined @FrameXML\MoneyFrame.lua:525
MoneyFrame_Update = <function> defined @FrameXML\MoneyFrame.lua:306
GetNumPrimaryProfessions = <function> defined =[C]:-1
GetTrainerServiceDescription = <function> defined =[C]:-1
StaticPopup_Visible = <function> defined @FrameXML\StaticPopup.lua:5061
StaticPopup_Hide = <function> defined @FrameXML\StaticPopup.lua:4739
```

**Comment by Xatrith** [comment]
Hello, can confirm it make the two window trample each other if you try to have your profession open while learning from the trainer.

**Comment by fusionpit** [comment]
@Xatrith sorry for not seeing your comment sooner!  That issue should be fixed in 0.6-beta and you will now be able to have a profession trainer and the skills screen open at the same time.

**Comment by fusionpit** [comment]
This should finally be resolved, in that you won't get an LUA error but instead will get an informational log printed to your chat window.
